statsmodels/gam/smooth_basis.py

```diff
@@ -792,6 +792,9 @@ def transform(self, x_new):
         basis : ndarray
             design matrix for the spline basis for given ``x_new``.
         """
+        x_new = np.asarray(x_new)
```

Comment:
You shouldn't case to array since x_new is expected to already be ndarray.

Reply:
Thanks for the comments! I think I've implemented all the requested changes. On this point, I've removed the offending line, but `AdditiveGamSmoother` supports the use of pandas DataFrames and Series at creation, so would adding support for them to `transform` be considered an enhancement?